src/common/buffer.cc

```diff
@@ -1945,7 +1945,8 @@ int buffer::list::write_fd(int fd) const
     return write_fd_zero_copy(fd);
 
   // use writev!
-  iovec iov[IOV_MAX];
+  int max = MIN(_buffers.size(), IOV_MAX);
```

Comment:
shouldn't we **assert(_buffers.size() <= IOV_MAX)** here instead ? It looks like the code below will overflow the array otherwise.